sherpa/astro/data.py

```diff
@@ -110,11 +110,11 @@
 this range to have at least 20 counts per group:
 
 >>> from sherpa.astro.io import read_pha
->>> pha = read_pha('src.pi')
+>>> pha = read_pha('sherpa/astro/datastack/tests/data/3c273.pi')
 >>> pha.subtract()
 >>> pha.set_analysis('energy')
 >>> pha.notice(0.5, 7)
->>> pha.group_counts(20, tabStops=~pha.mask)
```

Comment:
Why did the `tabStops` argument get removed?

Reply:
UNEXPECTED EXCEPTION: ValueError('grpNumCounts() The tabStops and countsArray have differing length')

Of course, I'm loading a specific, real data file so it might behave different now than the hypothetical `src.pi`. 
Throwing a few `get_filter_expr` in between, we see that, for this data file, even right after the notice call the filter boundary is a lot higher than requested because the bins are so wide. 
```python
In [8]: pha.get_filter_expr()
Out[8]: '0.0015-14.9504 Energy (keV)'

In [9]: pha.subtract()

In [10]: pha.set_analysis('energy')

In [11]: pha.notice(0.5, 7)

In [12]: pha.get_filter_expr()
Out[12]: '0.4672-9.8696 Energy (keV)'

In [13]: pha.group_counts(20)

In [14]: pha.get_filter_expr()
Out[14]: '0.4380-13.4612 Energy (keV)'
```
The root cause is that the data is read in grouped, and thus the length does not match even right after reading the file:
```python
In [24]: pha = read_pha(data_3c273 + '3c273.pi')
WARNING: systematic errors were not found in file 'sherpa/astro/datastack/tests/data/3c273.pi'
statistical errors were found in file 'sherpa/astro/datastack/tests/data/3c273.pi' 
but not used; to use them, re-read with use_errors=True
read ARF file sherpa/astro/datastack/tests/data/3c273.arf
read RMF file sherpa/astro/datastack/tests/data/3c273.rmf
WARNING: systematic errors were not found in file 'sherpa/astro/datastack/tests/data/3c273_bg.pi'
statistical errors were found in file 'sherpa/astro/datastack/tests/data/3c273_bg.pi' 
but not used; to use them, re-read with use_errors=True
read background file sherpa/astro/datastack/tests/data/3c273_bg.pi

In [25]: len(pha)
Out[25]: 1024

In [26]: len(pha.mask)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [26], in <cell line: 1>()
----> 1 len(pha.mask)

TypeError: object of type 'bool' has no len()

In [27]: pha.notice(0.5, 7)

In [28]: pha.mask.shape
Out[28]: (46,)
```

I have thus removed the `, tabStops=~pha.mask` to make an example that works in all cases and does not depend on loading a data file without initial grouping.

Reply:
This is one of those case where I was trying to let users know about the tabStops feature without a lot of fanfare. I think

    , tabstops=~pha.get_mask()

should work (as `get_mask` expands out the mask). Except when there's no filter present, but in this example we do have one.

```
>>> from sherpa.astro.io import read_pha
>>> pha = read_pha("sherpa-test-data/sherpatest/3c273.pi")
WARNING: systematic errors were not found in file 'sherpa-test-data/sherpatest/3c273.pi'
statistical errors were found in file 'sherpa-test-data/sherpatest/3c273.pi' 
but not used; to use them, re-read with use_errors=True
read ARF file sherpa-test-data/sherpatest/3c273.arf
read RMF file sherpa-test-data/sherpatest/3c273.rmf
WARNING: systematic errors were not found in file 'sherpa-test-data/sherpatest/3c273_bg.pi'
statistical errors were found in file 'sherpa-test-data/sherpatest/3c273_bg.pi' 
but not used; to use them, re-read with use_errors=True
read background file sherpa-test-data/sherpatest/3c273_bg.pi
>>> len(pha)
1024
>>> pha.mask
True
>>> pha.get_mask().shape
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'shape'
>>> pha.notice(0.5, 7)
>>> pha.mask.shape
(46,)
>>> pha.get_mask().shape
(1024,)
>>> pha.group_counts(20, tabStops=~pha.get_mask())
>>> 
```